internal/web_server/web/views/components/dashboard_components/user_admin/pagination.templ

```diff
@@ -0,0 +1,98 @@
+package user_admin
+
+import "fmt"
+import "strconv"
+
+// Pagination is the main component for the pagination
+templ Pagination(maxPages int, page int, parentTableID string, queryParam string) {
+	@prevButton(page, parentTableID)
+	@pageNumber(page, maxPages)
+	@nextButton(page, maxPages, parentTableID)
+	@insertHxGet(parentTableID, queryParam)
+}
+
+// handlePagination ensures that the page number is within the range of the maximum pages
+func handlePagination(page int, maxPages int) int {
+	if page < 1 {
+		page = 1
+	} else if page > maxPages {
+		page = maxPages
+	}
+	return page
+}
+
+// insertHxGet is a helper function that inserts the hx-get attribute to the pagination buttons
+script insertHxGet(parentTableID string, queryParam string) {
+	let params = new URL(document.location.toString()).searchParams;
+	let aPage = params.get("apage");
+	let nPage = params.get("npage");
+
+	if (aPage == null) {
+		aPage = 1;
+	}
+	if (nPage == null) {
+		nPage = 1;
+	}
+
+	let param = queryParam == "apage" ? aPage : nPage;
+
+	let oppositeQUery = queryParam == "apage" ? "npage" : "apage";
+	let oppositeParam = queryParam == "apage" ? nPage : aPage;
+	
+
+	const urlHold = "/api/v1/admin/username/table?"
+
+	var url = window.location.href.split('?')[0];
+
+	document.getElementById( `prev-button-${parentTableID}` ).setAttribute("href", url + "?" + queryParam + "=" + (parseInt(param) - 1) + "&" + oppositeQUery + "=" + oppositeParam);
```

Comment:
use the `` for string in the `setAttribute` instead of using string concatenation, for simplicity

Reply:
I dont how template literals are simpler, nor more readable. But done